src/Concordium/Client/Types/Contract/Info.hs

```diff
@@ -55,7 +60,7 @@ import Concordium.Client.Cli
 -- Logs fatally on internal errors that should never occur, namely:
 --  - Schema data has already been added.
 --  - The version of module schema and contract info does not match.
-addSchemaData :: ContractInfo -> CS.ModuleSchema -> ClientMonad IO (Maybe ContractInfo)
+addSchemaData :: (MonadIO m) => ContractInfo -> CS.ModuleSchema -> ClientMonad m (Maybe ContractInfo)
```

Comment:
Mostly out of curiosity. Should this function ideally not be pure? I don't see it doing any client or IO interaction.

Reply:
It does do logging to stdout/err. Or killing the process.

So it is effectful. I think it is perfectly sensible to be how it is. THis is ulimately about user interaction.